basis/db/tuples/tuples-docs.factor

```diff
@@ -122,33 +123,49 @@ HELP: update-tuple
      { "tuple" tuple } }
 { $description "Updates a tuple that has already been inserted into a database. The tuple must have a primary key that has been set by " { $link insert-tuple } " or that is user-defined." } ;
 
+HELP: map-tuples
+{ $values
+     { "query/tuple" tuple }
+     { "quot" quotation } }
+{ $description "An SQL query is constructed from the slots of the exemplar tuple that are not " { $link f } ". The " { $snippet "quot" } " is applied to each tuple from the database that matches the query, and the changed tuple is stored back to the database. If the " { $snippet "quot" } " returns " { $link f } ", the tuple is dropped, and its data remains unmodified in the database."
+$nl
+"The word is equivalent to the following code:"
+{ $code "query/tuple select-tuples quot map sift [ update-tuple ] each" }
+"The difference is that " { $snippet "map-tuples" } " handles query results one by one, thus avoiding the overhead of allocating the intermediate array of tuples, which " { $link select-tuples } " would do. This is important when processing large amounts of data in limited memory." } ;
+
 HELP: delete-tuples
 { $values
      { "tuple" tuple } }
 { $description "Uses the " { $snippet "tuple" } " as an exemplar object and deletes any objects that have the same slots set. If a slot is not " { $link f } ", then it is used to generate an SQL statement that deletes tuples." }
 { $warning "This word will delete your data." } ;
 
-{ insert-tuple update-tuple delete-tuples } related-words
+{ insert-tuple update-tuple map-tuples delete-tuples } related-words
+
+HELP: each-tuple
+{ $values
+     { "query/tuple" tuple }
+     { "quot" quotation } }
```

Comment:
You can put quotation stack effects in help.

``{ "quot" { $quotation ( tuple -- ) } }``

In a future version of Factor, we wouldn't need to redundantly describe what the system already knows if the actual stack effects are specified properly in the word definition.

Reply:
Believe it or not, I did try to include the stack effect into the help text, but failed to find the correct syntax. Thank you for the tip!